### PR TITLE
Api helper

### PR DESCRIPTION
**Pull request by dimahadgi**

### CODE REVIEW

File: api_conf.ini [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[api_connection_settings]
+login = empnetadmin@acme.com
+password = password
```
Comment (whichfinder) [inline api_conf.ini]: shouldn't be stored on github

File: api_conf.ini [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+auth_url = https://acme.qc2.blnsearch.net/auth/sessions
+post_url = https://acme.qc2.blnsearch.net/api-employer/certificates
+get_url =  https://acme.qc2.blnsearch.net/api-employer/workers?sort=created&order=desc&$limit=1000
```
Comment (whichfinder) [inline api_conf.ini]: set host url, path should be stored in method or class

File: src/for_testing.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+from src.utils.api_helper import ApiHelper
+
+api_help = ApiHelper()
+print(api_help.auth_token)
```
Comment (whichfinder) [inline src/for_testing.py]: add this as test, not as separate module

File: src/utils/api_helper.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    def auth_get_token(self):
+        response = requests.post(self.auth_url,
+                                 data={'grantType': "password_persistent",
+                                       'identifier': "{}".format(self.login),
+                                       'organizationKey': "acme",
+                                       'password': "{}".format(self.password)})
+        return response.json().get('token')
+
+    def do_get_request(self, get_url):
+        response = requests.get(get_url, headers={
```
Comment (whichfinder) [inline src/utils/api_helper.py]: create headers inside __init__

Comment (whichfinder) [reply src/utils/api_helper.py]: so you can use them as headers=self.headers in all requests

File: src/utils/certificates_generator.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    worker_id = dbconn.fetch_all('''select "id" from "worker" where "employerId"=201 and "archived"=false;''')
+    return [x[0] for x in worker_id]
+
+
+def auth_get_token():
+    response = requests.post('https://acme.qc2.blnsearch.net/auth/sessions',
+                             data={'grantType': "password_persistent",
+                                   'identifier': "admin@acme.com",
+                                   'organizationKey': "acme",
+                                   'password': "password"})
```
Comment (whichfinder) [inline src/utils/certificates_generator.py]: move to config

File: src/utils/certificates_generator.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    response = requests.get(GET_URL, headers={'Authorization': 'bln type=session,version=1,token="{}"'.format(auth_get_token())})
+    return response
+
+
+def generate_payload(workers_id):
+    body = {'courseName': "{}".format(fake.company()),
+            'description': "{}".format(fake.catch_phrase()),
+            'expiration': "{}-11-05T16:01:38.440Z".format(random.randint(2017, 2030)),
+            'file': "efb9c5a7-862b-46ca-9ce3-7c8110d0cbff_share rules.png",
+            'issued': "2017-11-05T16:01:38.433Z",
```
Comment (whichfinder) [inline src/utils/certificates_generator.py]: I think this is being changed from request to request

File: src/utils/certificates_generator.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
+import requests
+import random
+
+from src.utils.db_postgress_helper import DbConnect
+from faker import Faker
+fake = Faker('en_CA')
+
+POST_URL = "https://acme.qc2.blnsearch.net/api-employer/certificates"
+GET_URL = "https://acme.qc2.blnsearch.net/api-employer/workers?sort=created&order=desc&$limit=1000"
```
Comment (whichfinder) [inline src/utils/certificates_generator.py]: you've set the same in config

File: src/utils/certificates_generator.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+                                 headers={
+                                     'Authorization': 'bln type=session,version=1,token="{}"'.format(auth_get_token())},
+                                 json=generate_payload(workers_id))
+        if response.status_code == 201:
+            print('Certificate successfully added')
+        else:
+            print(response.status_code, generate_payload(workers_id))
+    return response.content
+
+do_post_request()
```
Comment (whichfinder) [inline src/utils/certificates_generator.py]: do this in if name = main

File: src/utils/certificates_generator.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+            'description': "{}".format(fake.catch_phrase()),
+            'expiration': "{}-11-05T16:01:38.440Z".format(random.randint(2017, 2030)),
+            'file': "efb9c5a7-862b-46ca-9ce3-7c8110d0cbff_share rules.png",
+            'issued': "2017-11-05T16:01:38.433Z",
+            'trainingProvider': "{}".format(fake.company()),
+            'workerId': workers_id}
+    return body
+
+
+def do_post_request():
```
Comment (whichfinder) [inline src/utils/certificates_generator.py]: this method should be generic not bounded to get_worker_id

**Review by whichfinder — CHANGES_REQUESTED** [review]

File: src/utils/api_helper.py [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import requests
+from src.config_parser import Config
+
+
+class ApiHelper:
+    def __init__(self):
+        self.auth_url = Config.api_test_host + Config.api_test_auth_path
```
Comment (whichfinder) [inline src/utils/api_helper.py]: os.path.join

File: src/for_testing.py [hunk 11]
```diff
@@ -0,0 +1,9 @@
+from src.utils.api_helper import ApiHelper
+#
+# api_help = ApiHelper()
+# print(api_help.auth_token)
+
+
+
+
+print(ApiHelper().auth_get_token())
```
Comment (whichfinder) [inline src/for_testing.py]: remove this module

File: src/utils/api_helper.py [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+        self.login = Config.api_test_login
+        self.password = Config.api_test_password
+        self.auth_token = self.auth_get_token()
+
+    def auth_get_token(self):
+        response = requests.post(self.auth_url,
+                                 data={'grantType': "password_persistent",
+                                       'identifier': "{}".format(self.login),
```
Comment (whichfinder) [inline src/utils/api_helper.py]: remove format, use values

File: src/utils/api_helper.py [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+                                       'organizationKey': "acme",
+                                       'password': "{}".format(self.password)})
```
Comment (whichfinder) [inline src/utils/api_helper.py]: same

File: src/utils/api_helper.py (hunk 4 above)
Comment (whichfinder) [reply src/utils/api_helper.py]: headers should be in init

File: src/utils/api_helper.py [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+        return response.json().get('token')
+
+    def do_get_request(self, get_url):
+        response = requests.get(get_url, headers={
+            'Authorization': 'bln type=session,version=1,token="{}"'.format(self.auth_token)})
+        return response
+
+    def do_post_request(self, post_url, body):
+        response = requests.post(post_url, headers={
+            'Authorization': 'bln type=session,version=1,token="{}"'.format(self.auth_token())}, json=body)
```
Comment (whichfinder) [inline src/utils/api_helper.py]: same here

File: src/utils/certificates_generator.py [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+                'description': self.fake.catch_phrase(),
+                'expiration': self.exp_date,
+                'file': self.image_file,
+                'issued': self.issued_date,
+                'trainingProvider': self.fake.company(),
+                'workerId': worker_id
+                }
+        return body
+
+    def add_certificates(self):
```
Comment (whichfinder) [inline src/utils/certificates_generator.py:45]: this method will be used inside tests, so it should return something, right?

File: src/utils/certificates_generator.py [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+        for workers_id in self.get_worker_id():
+            response = requests.post(self.url, headers=self.header, json=self.generate_payload(workers_id))
+            if response.status_code == 201:
+                print(datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S -"), 'Certificate successfully created')
+            else:
+                print(response.status_code)
```
Comment (whichfinder) [inline src/utils/certificates_generator.py]: remove print, 
We will use logger later

File: src/utils/certificates_generator.py [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+
+
+if __name__ == "__main__":
+    c = CertificatesGenerator()
+    c.add_certificates()
```
Comment (whichfinder) [inline src/utils/certificates_generator.py]: rename "c" to something meaningful

**Review by whichfinder — CHANGES_REQUESTED** [review]

File: src/for_testing.py (hunk 11 above)
Comment (dimahadgi) [reply src/for_testing.py]: added to git ignore

**Review by whichfinder — APPROVED** [review]

**Pull request merged**